File: lib/5_2/flutter/material/flatButton.dart
import 'package:flua/5_2/closure.dart';
import 'package:flua/5_2/context.dart';
import 'package:flua/5_2/flutter/maybeUnwrapAndBuildArgument.dart';
import 'package:flua/5_2/table.dart' as l;
import 'package:flutter/material.dart';

loadFlatButton(l.Table table) {
  table["flatButton"] = makeLuaDartFunc(func: (List<dynamic> args) {
    return [
      FlatButton(
        child: maybeUnwrapAndBuildArgument(args[0]["child"]),
        onPressed: () {
          Closure closure = args[0]["onPressed"];
          closure.call([]);
        },
      )
    ];
  });
}

File: runtime/flutter/material/flatButton.ts
import { RuntimeBaseClass } from "./../../runtimeBaseClass";
import { Type } from "../../dart/core/type";
import { StatelessWidget } from "./../widgets/statelessWidget";
import { Widget } from "../widget";

export interface FlatButtonProps {
    child: Widget;
    onPressed: (this: void) => void;
}

declare const flutter: {
    material: {
        flatButton: (this: void, props: FlatButtonProps) => FlatButton;
    };
};

export class FlatButton extends StatelessWidget implements RuntimeBaseClass {
    public readonly internalRuntimeType = new Type(FlatButton);
    public props: FlatButtonProps;
    public constructor(props: FlatButtonProps) {
        super();
        this.props = props;
    }

    public build(): Widget {
        return flutter.material.flatButton(this.props);
    }
}
